pkg/util/shortid_generator.go

```diff
@@ -25,7 +26,17 @@ func IsShortUIDTooLong(uid string) bool {
 	return len(uid) > 40
 }
 
-// GenerateShortUID generates a short unique identifier.
+// GenerateShortUID will generate a UUID that can also be a k8s name
+// it is gaurenteed to have a character as the first letter
+// This will return a valid k8s name
 func GenerateShortUID() string {
-	return shortid.MustGenerate()
+	uid, err := uuid.NewRandom()
+	if err != nil {
+		panic("invalid uuid")
```

Comment:
I realize we should never actually hit this, but would this panic be recovered somewhere if we did or would Grafana crash?

We had an implicit panic before in the form of `MustGenerate()` which was also very suboptimal, but it might be a good opportunity to rename this function, update it to return a proper error and update the callers to suit vs continuing the bad pattern.

Reply:
ya -- I left the panic because it exists in the `MustGenerate` -- I'll take a look at how bad bubbling up that error actually looks

Reply:
adding the error touches 200 things in 70 files.... given that the error is never really possible (i assume a LOT is going on if it is) i think we should keep the signature as is.

I updated it to fill random UUID bytes -- better than a panic I suppose?